### PR TITLE
Esc 295 promote other be as lead

### PR DESCRIPTION
**Pull request by priyankadhiman3006**
ticket details -> https://jira.tools.tax.service.gov.uk/browse/ESC-295

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala [hunk 1]
```diff
@@ -297,6 +297,16 @@ class BusinessEntityController @Inject()(
       case _ => handleMissingSessionData("Undertaking journey")
     }}
 
+  private def getNext(businessEntityJourney: BusinessEntityJourney)(implicit EORI: EORI): Future[Result] = {
+    businessEntityJourney.isLeadSelectJourney.contains(true) match {
+      case true =>  store.put[BusinessEntityJourney](BusinessEntityJourney(isLeadSelectJourney = true.some))
```
Comment (sam-miele) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:302]: NABD - 
```
businessEntityJourney.isLeadSelectJourney match {
   case Some(true) =>
    case _ =>
}
```

Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:302]: o yeah 👍

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/models/Undertaking.scala [hunk 2]
```diff
@@ -53,6 +53,8 @@ object Undertaking {
         .headOption
         .getOrElse(throw new IllegalStateException(s"BE with eori $eori is missing"))
     }
+
+    def getAllNonLeadEORIs(): List[EORI] = undertaking.undertakingBusinessEntity.filter(!_.leadEORI).map(_.businessEntityIdentifier)
```
[the remark this answers is not in the document]
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/models/Undertaking.scala:57]: sorry, didn't get you. Here  I am trying to get list of all non lead EORI's

[the remark this answers is not in the document]
Comment (sam-miele) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/models/Undertaking.scala:57]: Sorry. I misread something above! Ignore this. Setting to resolvd.

**Review by carwynellis — APPROVED** [review]
LGTM 👍

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala (hunk 1 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala]: updated [here](https://github.com/hmrc/eu-subsidy-compliance-frontend/pull/44/commits/ae89f21f7bc7e4daac5944fe14492445ddb4bd23)

**Review by sam-miele — APPROVED** [review]

**Pull request merged**